### PR TITLE
[Gutenberg] Upgrade React Native `0.71.11`

### PR DESCRIPTION
**Pull request by SiobhyB**
**Related to https://github.com/wordpress-mobile/gutenberg-mobile/pull/5874**

This PR contains the following changes:
* Bump Gutenberg Mobile reference to the latest commit of [GB-mobile branch `upgrade/react-native-0.71.8`](https://github.com/wordpress-mobile/gutenberg-mobile/pull/5874). This incorporates the needed changes for the RN upgrade coming from Gutenberg.
* Integration of Gutenberg using XCFramework (further information can be found in [this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/20717)).
* Fix for failing UI tests caused by the RN upgrade (further information can be found in [this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/21143)).

**To test:**
Follow testing instructions from https://github.com/WordPress/gutenberg/pull/51303.

## Regression Notes
1. Potential unintended areas of impact
None apart from the editor.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing and CI tests.

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:

- [x] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: Gutenberg/cocoapods_helpers.rb
Comment (fluiddot) [inline Gutenberg/cocoapods_helpers.rb]: On iOS we don't support Hermes yet, so I'm wondering if need to add this pod 🤔 .

**Comment by fluiddot** [comment]
Note that I've rebased this PR with changes from https://github.com/wordpress-mobile/WordPress-iOS/pull/20717 to incorporate the changes needed to integrate Gutenberg using the XCFramework.

**Comment by mokagio** [comment]
@fluiddot @SiobhyB, from the PR description at the time of writing:

> This PR contains the changes from https://github.com/wordpress-mobile/WordPress-iOS/pull/20717 and https://github.com/wordpress-mobile/WordPress-iOS/pull/21021, since the RN upgrade needs to use the XCFramework integration.

Should we close those PRs in favor of this one then? Or at least #21021, since #20717 has a few new commits with the work to add a build phase to detect nested frameworks from https://github.com/wordpress-mobile/WordPress-iOS/pull/21130 that haven't been cherry picked here. Presumably, once this PR lands the diff in #20717 will update with only those commits remaining.

**Comment by fluiddot** [comment]
> Should we close those PRs in favor of this one then? Or at least #21021, since #20717 has a few new commits with the work to add a build phase to detect nested frameworks from #21130 that haven't been cherry picked here. Presumably, once this PR lands the diff in #20717 will update with only those commits remaining.

Yep, I think we could use this PR as the main one, so makes sense to me to close those PRs. The idea is to incorporate all changes from #20717 and #21021, so I'll check if we need to rebase this branch to include everything. That said, I'll drop a message in each PR in this regard, and also include the reviewers from those here 👍 .

**UPDATE:** With https://github.com/wordpress-mobile/WordPress-iOS/pull/20956/commits/64811c63b9d78854f462dbb2b449eb606601481e, we should have all changes already incorporated in this branch.

File: Gutenberg/cocoapods_helpers.rb
Comment (fluiddot) [inline Gutenberg/cocoapods_helpers.rb:68]: These conditions will never be met as `gutenberg_dependencies` is only called when using Gutenberg locally, i.e. it will always reference the pod using a path.

File: Gutenberg/cocoapods_helpers.rb
Comment (fluiddot) [inline Gutenberg/cocoapods_helpers.rb]: As mentioned [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/21021#discussion_r1267610204), it's likely that we don't need to add `React-jsc` due to the use of Hermes.

File: Gutenberg/cocoapods_helpers.rb
Comment (fluiddot) [inline Gutenberg/cocoapods_helpers.rb]: If we confirmed that React Native preprends `$PWD`, I think we don't need to share details about the error. Hence, we could consider removing this part.

File: Scripts/BuildPhases/CopyGutenbergJS.sh
Comment (fluiddot) [inline Scripts/BuildPhases/CopyGutenbergJS.sh]: Typo:
```suggestion
# We check first for the XCFramework setup, falling back to source only if that is not there.
```

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

I shared some comments but none should be blocking the PR.

File: Gutenberg/cocoapods_helpers.rb
Comment (fluiddot) [reply Gutenberg/cocoapods_helpers.rb]: I confirmed that this is not needed when building using a local installation of Gutenberg, hence we can remove it.

File: Gutenberg/cocoapods_helpers.rb
Comment (fluiddot) [reply Gutenberg/cocoapods_helpers.rb:68]: Maybe it makes sense to keep it in case `gutenberg_dependencies` is used somewhere else in the future 🤔.

File: WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift
Comment (fluiddot) [inline WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift:659]: For some reason, when building the app using a local installation of Gutenberg, this line (and similar ones below) are giving the following warning when building: **Default will never be executed**

However, this is not the case when building using the XCFramework 🤔

File: Gutenberg/cocoapods_helpers.rb
Comment (fluiddot) [reply Gutenberg/cocoapods_helpers.rb]: I've applied this suggestion via https://github.com/wordpress-mobile/WordPress-iOS/pull/20956/commits/7e11b1e1a895173a05197dce99d6bbffdf43908f.

File: Gutenberg/cocoapods_helpers.rb
Comment (fluiddot) [reply Gutenberg/cocoapods_helpers.rb]: I've applied this update via https://github.com/wordpress-mobile/WordPress-iOS/pull/20956/commits/c992bcb76da42b26407b64a67f3bbc2ac37891a0.

**Review by dcalhoun — APPROVED** [review]

**Comment by fluiddot** [comment]
The CI jobs are failing with the following error after incorporating the latest changes from `trunk`:
```
❌ wordpress-ios/WordPress/Classes/ViewRelated/Media/MemoryCache.swift:4:8: no such module 'SDWebImage'
--
  | import SDWebImage
  | ^
```

The failure is related to [this PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/21120) and the usage of the library `SDWebImage` ([reference](https://github.com/wordpress-mobile/WordPress-iOS/blob/eb0588935f64252c67d5a254f8e4b96d425dbcb7/WordPress/Classes/ViewRelated/Media/MemoryCache.swift#L4)). This library is a transitive dependency of the `RNFastImage` library, which is only used by the Gutenberg editor.

https://github.com/wordpress-mobile/WordPress-iOS/blob/eb0588935f64252c67d5a254f8e4b96d425dbcb7/Gutenberg/cocoapods_helpers.rb#L57
https://github.com/wordpress-mobile/WordPress-iOS/blob/eb0588935f64252c67d5a254f8e4b96d425dbcb7/Podfile.lock#L447-L449

Now that Gutenberg is integrated via XCFramework, the `RNFastImage` library is no longer a dependency in the project ([reference](https://github.com/wordpress-mobile/WordPress-iOS/pull/20956/files#diff-e5b768ab2330ee29074f2eeadddd406f19e73b14673eb5f9b5ed6b5b77f4e363L447-L450)). Hence, we have to either re-incorporate it or stop using it. For now, I'll add it to the `Podfile` (https://github.com/wordpress-mobile/WordPress-iOS/pull/20956/commits/131057df3bd21abe20c7754dc51ae22b0772c17c) as I presume we need it but I'd appreciate feedback from the developers involved in the PR to confirm it (cc @kean @staskus @guarani). Thanks for the help 🙇 !

**Pull request merged**
